Pod/Classes/LSBinaryCodec.m

```diff
@@ -0,0 +1,324 @@
+            {
+                // map<string, string> baggage = 4; is equivalent on the wire to:
+                // message Entry {
+                //   string key = 1;
+                //   string value = 2;
+                // }
+                // repeated Entry = 4;
+
+                // Read the key, advance head.
+                KeyInfo mapKeyInfo = [LSBinaryCodec nextKeyFromProto:msg offset:&head];
+
+                NSString *key = [[NSString alloc] initWithData:[msg subdataWithRange:NSMakeRange(head, mapKeyInfo.length)] encoding:NSUTF8StringEncoding];
+
+                head += mapKeyInfo.length;
+
+                // Read the value, advance head.
+                KeyInfo mapValueInfo = [LSBinaryCodec nextKeyFromProto:msg offset:&head];
+                NSString *value = [[NSString alloc] initWithData:[msg subdataWithRange:NSMakeRange(head, mapValueInfo.length)] encoding:NSUTF8StringEncoding];
+
+                // Add it to our baggage dictionary.
+                [baggage setObject:value forKey:key];
+                break;
+            }
+            default:
+                // Ignore other fields.
+                break;
+        }
+    }
+
+    if (traceId == 0 || spanId == 0) {
+        if (errorPtr) {
+            *errorPtr = [NSError errorWithDomain:OTErrorDomain code:OTInvalidCarrierCode userInfo:nil];
+        }
+        return false;
+    }
+
+    // What if there was no baggage? Just an empty dict then, nbd.
+    [dest initWithTraceId:traceId spanId:spanId baggage:baggage];
+
+    return true;
+}
+
+
+// Reads the UInt64 out of the bytes presented in little endian form. Advances the offset past this number.
++ (UInt64)readLittleEndianUInt64From:(NSData *)data offset:(UInt64 *)offset {
+    UInt64 value = OSReadLittleInt64(data.bytes, *offset);
+    *offset += sizeof(UInt64);
+    return value;
+}
+
+// Reads the info for the next key in the protobuf file, advances offset to the start of the field
+// for this key.
+const UInt64 kFieldTypeMask = 0x7;
+
++ (KeyInfo)nextKeyFromProto:(NSData *)protoEnc offset:(UInt64 *)offset {
+    UInt64 varint = [LSBinaryCodec readVarintFromProto:protoEnc startingAt:offset];
+
+    KeyInfo keyInfo;
+    keyInfo.fieldNum =  varint >> 3;
+    keyInfo.fieldType = varint & kFieldTypeMask;
+
+    switch (keyInfo.fieldType) {
+        case WireFormatVarint:
+            keyInfo.length = 0;
+            break;
+        case WireFormatFixed64:
+            keyInfo.length = sizeof(UInt64);
+            break;
+        case WireFormatFixed32:
+            keyInfo.length = sizeof(UInt32);
+            break;
+        case WireFormatLengthDelim:
+        {
+            UInt64 length = [LSBinaryCodec readVarintFromProto:protoEnc startingAt:offset];
+            keyInfo.length = length;
+            break;
+        }
+        default:
+            break;
+    }
+
+    return keyInfo;
+}
+
+// Reads a varint off the proto bytes, advances the offset to the next byte after the varint.
```

Comment:
assuming you are basing each of these helper functions on C/C++ code that lives elsewhere, can you link to those function definitions in each function comment?